core/src/main/scala/kafka/security/authorizer/AclAuthorizer.scala

```diff
@@ -164,7 +163,7 @@ class AclAuthorizer extends Authorizer with Logging {
       }.groupBy(_._1.pattern)
 
     if (aclsToCreate.nonEmpty) {
-      inWriteLock(lock) {
+      lock synchronized {
```

Comment:
`lock` is currently a `ReentrantReadWriteLock`. We could make it an Object or synchronize on the authorizer instance.

Reply:
Good point. Thanks for picking this up. I'll synchronize on the authorizer instance.